### PR TITLE
[WIP] Event updates and other miscellaneous changes

### PR DESCRIPTION
**Pull request by 0xilly**

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    /**
+     * Get a light value for this block, taking into account the given state and coordinates, normal ranges are between 0 and 15
+     *
+     * @param state
+     * @param reader
+     * @param pos
+     * @return The light value
+     */
+    default int getLightVaule(IBlockState state, IWorldReader reader, BlockPos pos)
```
Comment (Senmori) [inline src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java]: Typo?

Comment (0xilly) [reply src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java]: yep will fix in my next commit

**Comment by Senmori** [comment]
I noticed a few typos.

`default int getLightVaule(...)` in `IForgeBlock` (already addressed)
`hasTIleEntity(..)` in `IForgeBlock`

As well, maybe rename the following to be more generic and descriptive?
`isBurning(IWorldReader, BlockPos)` in `IForgeBlock` to `canIgnite` or something similar to better describe what is supposed to happen. Personally, `isBurning` would tell a user if the block is currently burning.

`isLadder(IBlockState, IWorldReader, BlockPos, EntityLivingBase)` in `IForgeBlock`
Possibly rename to `canClimb` for the same reasons as above?

**Comment by pau101** [comment]
@Senmori But `isBurning` is supposed to indicate if the block is burning. (well really `isFiery` could be more accurate). However, `isLadder` does warrant a rename and a more appropriate one would be `isClimbable`. The verb "can" is generally more appropriate when speaking about an attribute of an external object.

**Comment by Senmori** [comment]
@pau101 Yeah, I re-read the documentation and you're right. `canIgnite` would be wrong.

**Pull request merged**